### PR TITLE
Fixed null deprecation array_keys in Multishipping.

### PR DESCRIPTION
**Pull request by kiatng**

### CODE REVIEW

File: app/code/core/Mage/Core/Model/Session.php [hunk 1]
```diff
@@ -75,4 +74,9 @@ public function validateFormKey($formKey)
     {
         return ($formKey === $this->getFormKey());
     }
+
+    public function getOrderIds($clear = false): array
```
Comment (sreichel) [inline app/code/core/Mage/Core/Model/Session.php:78]:
```suggestion
    public function getOrderIds(bool $clear = false): array
```

**Comment by kiatng** [comment]
> Reproducable in core?

Yes, it happens:
Uncaught TypeError: array_keys(): Argument #1 ($array) must be of type array, null given in .../app/code/core/Mage/Checkout/Model/Type/Multishipping.php:618
Stack trace:
#0 .../app/code/core/Mage/Checkout/Model/Type/Multishipping.php(618): array_keys()
#1 .../app/code/core/Mage/Checkout/controllers/MultishippingController.php(578): Mage_Checkout_Model_Type_Multishipping->getOrderIds()
#2 .../app/code/core/Mage/Core/Controller/Varien/Action.php(419): Mage_Checkout_MultishippingController->successAction()
#3 .../app/code/core/Mage/Core/Controller/Varien/Router/Standard.php(255): Mage_Core_Controller_Varien_Action->dispatch()
#4 .../app/code/core/Mage/Core/Controller/Varien/Front.php(180): Mage_Core_Controller_Varien_Router_Standard->match()
#5 .../app/code/core/Mage/Core/Model/App.php(358): Mage_Core_Controller_Varien_Front->dispatch()
#6 .../app/Mage.php(748): Mage_Core_Model_App->run()
#7 .../index.php(68): Mage::run()

**Comment by sreichel** [comment]
I am not sure about adding methods for core/session. Better fix it in Mage_Checkout_Model_Type_Multishipping?

**Comment by kiatng** [comment]
My opinion is fixing it upstream is better, it's fixing at one source. It'll fix 3rd-party code.

**Comment by sreichel** [comment]
What the reason for clear parameter?

**Comment by kiatng** [comment]
See https://github.com/OpenMage/magento-lts/blob/c9a901f392326e7d76d10fbc17693a83915edbe0/app/code/core/Mage/Core/Model/Session/Abstract/Varien.php#L299-L306

And 

https://github.com/OpenMage/magento-lts/blob/c9a901f392326e7d76d10fbc17693a83915edbe0/app/code/core/Mage/Checkout/Block/Multishipping/Success.php#L30-L30

`$clear` is traditionally used to unset the `$_data`.

**Review by sreichel — APPROVED** [review]

**Review by addison74 — APPROVED** [review]

**Pull request merged**